agent/engine/docker_task_engine.go

```diff
@@ -1714,3 +1747,13 @@ func (engine *DockerTaskEngine) getDockerID(task *apitask.Task, container *apico
 	}
 	return dockerContainer.DockerID, nil
 }
+
+// getTaskENI returns the primary eni of the task.
+func (engine *DockerTaskEngine) getTaskENI(task *apitask.Task) *apieni.ENI {
```

Comment:
it seems to me that this part of logic should better be in task.go instead of here. if the current implementation of [GetPrimaryENI](https://github.com/aws/amazon-ecs-agent/blob/2874131a5f3d1c0908e0f2ebad62e48b8ff5850f/agent/api/task/task.go#L2148) does not work for windows, can that method have an alternative impl for windows and then we just use task.GetPrimaryENI() in the engine?

Reply:
Yes that works for Windows as well. I'll make these changes.